server/src/main/java/org/elasticsearch/action/support/IndexComponentSelector.java

```diff
@@ -23,14 +24,11 @@
  * We define as index components the two different sets of indices a data stream could consist of:
  * - DATA: represents the backing indices
  * - FAILURES: represent the failing indices
- * - ALL: represents all available in this expression components, meaning if it's a data stream both backing and failure indices and if it's
- * an index only the index itself.
  * Note: An index is its own DATA component, but it cannot have a FAILURE component.
  */
 public enum IndexComponentSelector implements Writeable {
     DATA("data", (byte) 0),
-    FAILURES("failures", (byte) 1),
-    ALL_APPLICABLE("*", (byte) 2);
+    FAILURES("failures", (byte) 1);
```

Comment:
This is an unlikely scenario but could we ever end up in a place where a new selector added here could collide with the old `::*` value in a mixed cluster?

Reply:
Hm.... Interesting thought. Thinking out loud because this is a tough cookie. 

If we implement this the way it's now and backport it back to `8.18` and `8.x`.
We can remove this backwards compatibility code from `9.x`; because you always have to go through the `8.last` version. `8.last` will never propagate this selector because it matches it to `::data` and this selector is never persisted anywhere, I think. Furthermore, there is no CCR or CCS supported yet. Right?

This means that if we ever introduced `::*` in `9+` we could distinguish the two. Right?